src/core/state.ts

```diff
@@ -172,6 +205,32 @@ export class State {
     return Bluebird.fromCallback(cb => this.cookieJar['_jar'].serialize(cb));
   }
 
+  public async exportState(withConstants: boolean = false): Promise<string> {
```

Comment:
The semantic meaning of this function is usually called the word serialization. Plus `state.exportState` looks overloaded, you alreaty wrote `state` once.
I offer to follow the best practices and call it `serialize` and `deserialize`.

Reply:
Additionally it definitely should return object instead of string because the most popular use case is to persist the state in mongodb or RDBMS json type field.

Reply:
One more thing - this function should not accept args like `withConstants: boolean = false`. Because why exactly constants? Why not device or cookies? Or timezoneOffset 😄 ? 
And the function call looks unobvious `igpapi.state.serialize(false)` - need to go to function definition to remind what that `false` means. 
It's up to user to write

```ts
const state = await igpapi.state.serialize();
delete state.constants;
```

Much more obvious for user, less code in lib implementation.